cpmpy/expressions/globalconstraints.py

```diff
@@ -582,3 +584,61 @@ def deepcopy(self, memodict={}):
         return Cumulative(*copied_args)
 
 
+class GlobalCardinalityCount(GlobalConstraint):
+    """
+        GlobalCardinalityCount(a,gcc): Collect the number of occurrences of each value 0..a.ub in gcc.
+    The array gcc must have elements 0..ub (so of size ub+1).
+        """
+
+    def __init__(self, a, gcc):
+        super().__init__("gcc", [a,gcc])
+
+    def decompose(self):
+        a, gcc = self.args
+        ub = max([v.ub for v in a])
+        assert (len(gcc) == ub + 1), f"GCC: length of gcc variables {len(gcc)} must be ub+1 {ub + 1}"
+        return [Count(a, i) == v for i, v in enumerate(gcc)]
+
+    def value(self):
+        a, gcc = self.args
+        gval = [argval(y) for y in gcc]
+        return all([gval[i] == Count(a,i).value() for i in range(len(gcc))])
```

Comment:
I guess you could do `all(self.decompose()).value()`? Would be cleaner? or not possible?

Reply:
Yes that works, changed it!